### PR TITLE
Tempo: Upgrade trace to profiles docs

### PR DESCRIPTION
**Pull request by joey-grafana**
**What is this feature?**

Adds section for embedded flame graphs, updated screenshots, provisioning trace to profiles and a table to explain the trace to profiles configuration options.

**Why do we need this feature?**

More complete documentation.

**Who is this feature for?**

Tempo / Pyroscope users.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/shared/datasources/tempo-traces-to-profiles.md [hunk 1]
```diff
@@ -28,11 +28,15 @@ There are two ways to configure the trace to profiles feature:
...
 
-![Selecting a link in the span queries the profile data source](/static/img/docs/tempo/profiles/tempo-profiles-Span-link-profile-data-source.png)
+{{< figure src="/static/img/docs/tempo/profiles/tempo-trace-to-profile.png" max-width="900px" class="docs-image--no-shadow" alt="Selecting a link in the span queries the profile data source" >}}
 
 To use trace to profiles, you must have a configured Grafana Pyroscope data source. For more information, refer to the [Grafana Pyroscope data source documentation](/docs/grafana/latest/datasources/grafana-pyroscope/).
 
-## Use a simple configuration
+**Embedded flame graphs** are also inserted into each span details section that has a linked profile (once you have configured a Grafana Pyroscope data source). This allows you to see resource consumption in a flame graph visualisation for each span without having to navigate away from the current view. Hover over a particular block in the flame graph to see more details about the resources being consumed.
```
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-traces-to-profiles.md:35]:
```suggestion
**Embedded flame graphs** are also inserted into each span details section that has a linked profile (requires a  configured Grafana Pyroscope data source). This lets you see resource consumption in a flame graph visualization for each span without having to navigate away from the current view. Hover over a particular block in the flame graph to see more details about the resources being consumed.
```

File: docs/sources/shared/datasources/tempo-traces-to-profiles.md [hunk 2]
```diff
@@ -28,11 +28,15 @@ There are two ways to configure the trace to profiles feature:
...
+
+In order to determine that there is a profile for a given span and render the **Profiles for this span** button or an embedded flame graph in the span details, the ‘pyroscope.profile.id’ key-value pair must exist in your span tags.
```
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-traces-to-profiles.md:37]: How does someone tell if the `pyroscope.profile.id` key-value pair exists in their span tags? Is this a query they can run in the Tempo data source query editor or some other easy way? 

I'm stumbling over the first section of the sentence. Is this correct? 
* If the pyorscope.profile.id key value pair is not present, then no profile data is present for a given span. 
* In that case, the Profiles for spans button is unavailable(?) and the flame graph doesn't appear. 

Is that correct? If so, would something like this be helpful:  

> If you have configured a Pyroscope data source and no profile data is available  or the Profiles for this span button is greyed out, verify that the `pyroscope.profile.id` key-value pair exists in your span tags.

File: docs/sources/explore/trace-integration.md [hunk 3]
```diff
@@ -120,7 +120,7 @@ You can navigate from a span in a trace view directly to metrics relevant for th
 Using Trace to profiles, you can use Grafana’s ability to correlate different signals by adding the functionality to link between traces and profiles.
 Refer to the [relevant documentation](/docs/grafana/latest/datasources/tempo/configure-tempo-data-source#trace-to-profiles) for configuration instructions.
 
-![Selecting a link in the span queries the profile data source](/static/img/docs/tempo/profiles/tempo-profiles-Span-link-profile-data-source.png)
+{{< figure src="/static/img/docs/tempo/profiles/tempo-trace-to-profile.png" max-width="900px" class="docs-image--no-shadow" alt="Selecting a link in the span queries the profile data source" >}}
```
Comment (knylander-grafana) [inline docs/sources/explore/trace-integration.md:123]:
```suggestion
{{< figure src="/static/img/docs/tempo/profiles/tempo-trace-to-profile.png" max-width="900px" class="docs-image--no-shadow" alt="Select a link in the span queries the profile data source" >}}
```

File: docs/sources/shared/datasources/tempo-traces-to-profiles.md [hunk 4]
```diff
@@ -78,3 +82,15 @@ To use a variable you need to wrap it in `${}`. For example, `${__span.name}`.
...
 | **\_\_trace.duration** | The duration of the trace.                                                                                                                                                                                                                                                                                                               |
 | **\_\_trace.name**     | The name of the trace.                                                                                                                                                                                                                                                                                                                   |
+
+## Configure trace to profiles
+
+The following table describes the ways in which you can configure your trace to profiles settings:
```
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-traces-to-profiles.md:88]:
```suggestion
The following table describes options for configuring your trace to profiles settings:
```

**Review by knylander-grafana — APPROVED** [review]
Approving with a few minor things. There is one sentence I had trouble parsing. As always, thank you for updating docs.

File: docs/sources/shared/datasources/tempo-traces-to-profiles.md (hunk 2 above)
Comment (joey-grafana) [reply docs/sources/shared/datasources/tempo-traces-to-profiles.md:37]: That is correct, will add the changes. Thanks, Kim :)

**Pull request merged**